storm-core/src/jvm/org/apache/storm/topology/WindowedBoltExecutor.java

```diff
@@ -76,7 +76,7 @@ public WindowedBoltExecutor(IWindowedBolt bolt) {
         this.bolt = bolt;
     }
 
-    private int getTopologyTimeoutMillis(Map stormConf) {
+    private long getTopologyTimeoutMillis(Map stormConf) {
```

Comment:
finally , the return value is `timeout * 1000` , it have convert to millisecond . Long is a better choose .